libs/pika/async_mpi/src/mpi_future.cpp

```diff
@@ -226,150 +357,95 @@ namespace pika { namespace mpi { namespace experimental {
             static auto poll_deb =
                 mpi_debug.make_timer(1, debug::str<>("Poll - lock success"));
             // output mpi debug info every N seconds
-            mpi_debug.timed(poll_deb, detail::get_mpi_info());
+            mpi_debug.timed(poll_deb, detail::mpi_data_);
         }
 
+        // Before moving requests from the queue to the vector
+        compact_vectors();
```

Comment:
not really, but I discovered some quite long vectors when throttling was not enabled and when adding from the queue to the vector, it makes the tracking of sizes easier, if I already know there are no `NULL_REQUESTS` in the vector (and you never need to scan the vector and `count_if` since tou always compact before adding. Since I removed one vector and use some atomics to track sizes/etc I decided this would be cleaner/easier.